cli/cmd/util.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/cncf/servicebroker/k8s/service_controller/model"
+)
+
+// from: https://play.golang.org/p/4FkNSiUDMg
+// newUUID generates a random UUID according to RFC 4122
+func NewUUID() (string, error) {
+	uuid := make([]byte, 16)
+	n, err := io.ReadFull(rand.Reader, uuid)
+	if n != len(uuid) || err != nil {
+		return "", err
+	}
+	// variant bits; see section 4.1.1
+	uuid[8] = uuid[8]&^0xc0 | 0x80
+	// version 4 (pseudo-random); see section 4.1.3
+	uuid[6] = uuid[6]&^0xf0 | 0x40
+	return fmt.Sprintf("%x-%x-%x-%x-%x", uuid[0:4], uuid[4:6], uuid[6:8], uuid[8:10], uuid[10:]), nil
+}
+
+func fetchInventory() (*model.Catalog, error) {
+	u := fmt.Sprintf("%s%s", controller, SERVICE_PLANS_URL)
+	i, err := callHttp(u, "GET", "inventory", nil)
+	if err != nil {
+		return nil, err
+	}
+	var c model.Catalog
+	err = json.Unmarshal([]byte(i), &c)
+	if err != nil {
+		return nil, err
+	}
+	return &c, nil
+}
+
+// Fetches the inventory from the SC and maps service:plan to the unique ID of the service plan
+// This could be more efficient with client side caching, etc. but for now will suffice.
+func fetchServicePlanGUID(service string, plan string) (string, error) {
+	u := fmt.Sprintf("%s%s", controller, SERVICE_PLANS_URL)
+	i, err := callHttp(u, "GET", "inventory", nil)
+	if err != nil {
+		return "", err
+	}
+	var c model.Catalog
+	err = json.Unmarshal([]byte(i), &c)
+	if err != nil {
+		return "", err
+	}
+	for _, s := range c.Services {
```

Comment:
Can you put a TODO here to move this into a library? We have a ton of these methods that get a plan, or plan ID, or service, given a name and a catalog.